src/librustc_metadata/encoder.rs

```diff
@@ -1321,10 +1327,11 @@ impl EncodeContext<'tcx> {
     fn encode_info_for_closure(&mut self, def_id: DefId) {
         debug!("EncodeContext::encode_info_for_closure({:?})", def_id);
 
-        let tables = self.tcx.typeck_tables_of(def_id);
+        // FIXME(eddyb) would `tcx.type_of(def_id)` work as well?
```

Comment:
Can you try that? It looks like it should.

Reply:
I suppose it's easy enough to check.